### PR TITLE
Add a stub for NativeMethodHandle in JDK16

### PR DESCRIPTION
**Pull request by babsingh**
fixes: https://github.com/eclipse/openj9/issues/11255

Signed-off-by: Babneet Singh <sbabneet@ca.ibm.com>

### CODE REVIEW

**Comment by babsingh** [comment]
I tried to turn-on the `opt_panama` and `Panama` (JCL) flags in order to enable the OpenJ9 `NativeMethodHandle` support. But, I ended up facing many issues:
1. `jdk.internal.nicl` and `java.nicl` packages no longer exist in JDK16. Finding their new equivalent representation will take some time. 
2. `MethodHandles` class no longer has the `findNative*` methods.
3. `Library` class is no longer available; potentially renamed to `NativeLibrary` and relocated.
4. `gcc` error: `Jit.hpp` in `JavaMethodCallbackGen.cpp` cannot be found.
5. ... the list goes on

The above conflicts cannot be resolved quickly. So, I have added a minimal stub for the `NativeMethodHandle` class, which should allow us to quickly enable JDK16 builds.

**Comment by babsingh** [comment]
Suggested PR builds:
```
Jenkins test sanity xlinux jdk11
Jenkins test sanity xlinux jdknext
Jenkins test zlinux jdknext depends ibmruntimes/openj9-openjdk-jdk#openj9-staging
```
All builds should pass.

**Comment by babsingh** [comment]
fyi @tajila @JasonFengJ9 @pshipton @gacholio @keithc-ca

**Comment by tajila** [comment]
Jenkins test sanity xlinux jdk11

**Comment by tajila** [comment]
Jenkins test sanity plinux jdknext

**Comment by tajila** [comment]
Jenkins test sanity zlinux jdknext depends ibmruntimes/openj9-openjdk-jdk#openj9-staging

File: jcl/src/java.base/share/classes/java/lang/invoke/NativeMethodHandle.java [hunk 1]
```diff
@@ -112,3 +113,25 @@ void compareWithNative(NativeMethodHandle left, Comparator c) {
...
 	}
 }
+/*[ELSE]*/
+import java.util.List;
+import jdk.internal.invoke.NativeEntryPoint;
+
+class NativeMethodHandle extends MethodHandle {
+	public static MethodHandle make(NativeEntryPoint entryPoint, MethodHandle mh) {
+		OpenJDKCompileStub.OpenJDKCompileStubThrowError();
+		return null;
```
Comment (keithc-ca) [inline jcl/src/java.base/share/classes/java/lang/invoke/NativeMethodHandle.java:123]: All uses of `OpenJDKCompileStubThrowError()` should be in a `throw` statement:
```
  throw OpenJDKCompileStub.OpenJDKCompileStubThrowError();
```

File: jcl/src/java.base/share/classes/java/lang/invoke/NativeMethodHandle.java [hunk 2]
```diff
@@ -112,3 +113,25 @@ void compareWithNative(NativeMethodHandle left, Comparator c) {
 		c.compareStructuralParameter(left.vmSlot, this.vmSlot);
 	}
 }
+/*[ELSE]*/
```
Comment (keithc-ca) [inline jcl/src/java.base/share/classes/java/lang/invoke/NativeMethodHandle.java:116]: This should repeat the condition:
```
/*[ELSE] Panama */
```

File: jcl/src/java.base/share/classes/java/lang/invoke/NativeMethodHandle.java [hunk 3]
```diff
@@ -1,4 +1,4 @@
-/*[INCLUDE-IF Panama]*/
+/*[INCLUDE-IF Panama|(JAVA_SPEC_VERSION >= 16)]*/
```
Comment (keithc-ca) [inline jcl/src/java.base/share/classes/java/lang/invoke/NativeMethodHandle.java:1]: Please use spaces around binary operators, just like you would in C code:
```
/*[INCLUDE-IF Panama | (JAVA_SPEC_VERSION >= 16)]*/
```

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: jcl/src/java.base/share/classes/java/lang/invoke/NativeMethodHandle.java [hunk 4]
```diff
@@ -112,3 +113,25 @@ void compareWithNative(NativeMethodHandle left, Comparator c) {
...
 	}
 }
+/*[ELSE] Panama */
+import java.util.List;
+import jdk.internal.invoke.NativeEntryPoint;
+
+class NativeMethodHandle extends MethodHandle {
+	public static MethodHandle make(NativeEntryPoint entryPoint, MethodHandle mh) {
+		throw OpenJDKCompileStub.OpenJDKCompileStubThrowError();
+		return null;
```
Comment (keithc-ca) [inline jcl/src/java.base/share/classes/java/lang/invoke/NativeMethodHandle.java:123]: This won't compile: the `return` here (and in `addRelatedMHs()`) is unreachable.

**Comment by babsingh** [comment]
All the recent feedback has been addressed. Unfortunately, `jdknext openj9` build will be unsuccessful since `NativeEntryPoint` only exists in `jdknext openj9-staging` at this point.

**Comment by keithc-ca** [comment]
Jenkins test sanity xlinux jdk11

**Comment by keithc-ca** [comment]
Jenkins test sanity zlinux jdknext depends ibmruntimes/openj9-openjdk-jdk#openj9-staging

**Review by keithc-ca — APPROVED** [review]

**Pull request merged**